src/Form.elm

```diff
@@ -197,14 +200,57 @@ update validation msg (F model) =
                     else
                         model.dirtyFields
 
-                newChangedFields =
-                    Set.insert name model.changedFields
+                ( newChangedFields, newOriginalValues ) =
+                    if Set.member name model.changedFields then
+                        let
+                            storedValue =
+                                Dict.get
+                                    name
+                                    model.originalValues
+                                    |> Maybe.withDefault Nothing
```

Comment:
`Dict.get` will return `Nothing` so the `|> Maybe.withDefault Nothing` is not needed.